core/src/apps/solana/token_account.py

```diff
@@ -0,0 +1,65 @@
+from typing import TYPE_CHECKING
+
+from trezor.crypto import base58
+
+if TYPE_CHECKING:
+    from trezor.messages import SolanaTxTokenAccountInfo
+
+ASSOCIATED_TOKEN_ACCOUNT_PROGRAM = "ATokenGPvbdGVxr1b2hvZbsiqW5xWH25efTNsLJA8knL"
+
+SEED_CONSTANT = "ProgramDerivedAddress"
+
+
+def assert_is_associated_token_account(
+    base_address: bytes,
+    token_account_address: bytes,
+    token_program: bytes,
+    token_mint: bytes,
+) -> None:
+    from trezor.crypto.hashlib import sha256
+
+    # based on the following sources:
+    # https://spl.solana.com/associated-token-account#finding-the-associated-token-account-address
+    # https://github.com/solana-labs/solana-web3.js/blob/745437f79bf4e268929e7740cfc517a2fcee980f/packages/library-legacy/src/publickey.ts#L212
+    for seed_bump in range(255, 0, -1):
```

Comment:
none of the sources tell me why seed_bump should start at 255 and go down...?

Reply:
I haven't found a proper source describing this other than the ones posted below - I'll add those to the code.

Reply:
Maybe https://solanacookbook.com/core-concepts/pdas.html#generating-pdas.